src/wrapper/pack.rs

```diff
@@ -22,24 +24,48 @@ pub fn pack_image_cmd(wrapper: &Wrapper, cmdline: Vec<String>)
     let cconfig = try!(wrapper.config.containers.get(&options.name)
         .ok_or(format!("Container {} not found", options.name)));
     let container_ver = wrapper.root.as_ref().unwrap();
-    let root = Path::new("/vagga/base/.roots")
-        .join(container_ver).join("root");
 
-    let mut cmd = Command::new("/vagga/bin/busybox");
-    cmd.stdin(Stdio::null())
+    let tmppath = PathBuf::from(
+        &format!("/vagga/base/.roots/.tmp.{}", &options.name));
+    match build::prepare_tmp_root_dir(&tmppath) {
+        Ok(()) => {}
+        Err(x) => {
+            return Err(format!("Error preparing root dir: {}", x));
+        }
+    }
+
+    let mut capsule_features = vec!();
+    let root = Path::new("/vagga/base/.roots")
+        .join(container_ver)
+        .join("root");
+    let mut tar_cmd = Command::new("/vagga/bin/busybox");
+    tar_cmd.stdin(Stdio::null())
         .arg("tar")
         .arg("-c");
-    set_uidmap(&mut cmd,
-        &try!(map_users(wrapper.settings, &cconfig.uids, &cconfig.gids)),
-        false);
+    if let Some(compression_type) = options.compression_type {
+        tar_cmd.arg(compression_type.get_short_option());
+        capsule_features.push(compression_type.get_capsule_feature());
+    }
     if let Some(ref f) = options.file {
-        cmd.arg("-f").arg(Path::new("/work").join(f));
+        tar_cmd.arg("-f");
+        if f.starts_with("/") {
+            tar_cmd.arg(f);
+        } else {
+            tar_cmd.arg(Path::new("/work").join(f));
+        }
+    }
+    tar_cmd
+        .arg("-C").arg(&root)
+        .arg(".");
+
+    if capsule_features.len() > 0 {
+        let mut ctx = BuilderContext::new(&wrapper.config, options.name.clone(),
```

Comment:
Looks like huge hack. I think we could break down `ensure_features` and pass just `capsule::State` and settings instead of creating a build context.